ext/standard/math.c

```diff
@@ -1024,15 +1024,30 @@ PHPAPI zend_string *_php_math_number_format_ex(double d, int dec, const char *de
 	size_t reslen = 0;
 	int count = 0;
 	int is_negative=0;
+	int dec_round;
+
+	// prevent integer overflow
+#if SIZEOF_ZEND_LONG > SIZEOF_INT
+	if (dec >= 0) {
+		dec_round = dec > INT_MAX ? INT_MAX : (int)dec;
+	} else {
+		dec_round = dec <= INT_MIN ? INT_MIN+1 : (int)dec;
```

Comment:
Why is there a `+1` for the min case?

Reply:
Good question - I took it from `PHP_FUNCTION(round)` - https://github.com/php/php-src/blob/master/ext/standard/math.c#L286-L294 and I don't know the answer :see_no_evil:

Reply:
Ah, `_php_math_round` calls `abs` with this number. Although it also already has a check for this:

https://github.com/php/php-src/blob/69a8b63ecf4fec1b35ef4da1ac9579321c45f97f/ext/standard/math.c#L135

Reply:
I have removed the `+1` in base cases as it's not needed but confusion